src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (Object.keys(coords).length === 0) {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    };
```

Comment:
Check using `Object.keys` is not the most reliable solution.
Coords may have one attribute, and this code will not work correctly.
Here's more stable and simple version:
```suggestion
this.coords = {
  x: coords.x || 0,
  y: coords.y || 0,
};
```